lib/modules/platform/github/index.ts

```diff
@@ -241,6 +242,87 @@ export async function getJsonFile(
   return JSON5.parse(raw);
 }
 
+export async function getForkOrgs(token: string): Promise<string[]> {
+  // This function will be adapted later to support configured forkOrgs
+  if (!config.renovateForkUser) {
+    try {
+      logger.debug('Determining fork user from API');
+      const userDetails = await getUserDetails(platformConfig.endpoint, token);
+      config.renovateForkUser = userDetails.username;
+    } catch (err) {
+      logger.debug({ err }, 'Error getting username for forkToken');
+    }
+  }
+  return config.renovateForkUser ? [config.renovateForkUser] : [];
+}
+
+export async function listForks(
+  token: string,
+  repository: string
+): Promise<GhRestRepo[]> {
+  // Get list of existing repos
+  const url = `repos/${repository}/forks?per_page=100`;
+  const repos =
+    (
+      await githubApi.getJson<GhRestRepo[]>(url, {
+        token,
+        paginate: true,
+        pageLimit: 100,
+      })
+    )?.body ?? [];
+  logger.debug(`Found ${repos.length} forked repo(s)`);
+  return repos;
+}
+
+export async function findFork(
+  token: string,
+  repository: string
+): Promise<GhRestRepo | null> {
+  const forks = await listForks(token, repository);
+  const orgs = await getForkOrgs(token);
+  if (!orgs.length) {
+    throw new Error(REPOSITORY_CANNOT_FORK);
+  }
+  let forkedRepo: GhRestRepo | undefined;
+  for (const forkOrg of orgs) {
+    logger.debug(`Searching for forked repo in ${forkOrg}`);
+    forkedRepo = forks.find((repo) => repo.owner.login === forkOrg);
```

Comment:
forks can be a huge list, so better use a map where key is the owner login.

Reply:
Do you think it really takes that much time to iterate through?

Reply:
depends a little bit of the length of the outer loop. for a single org it's no problem, but for 3 org you iterate three time in worst case. so its getting more worse with more orgs

Reply:
It should be at most 2 orgs. Also, I can't easily replicate more than 1 org in this PR, so I'd like to push this to the next PR if necessary